index.js

```diff
@@ -119,243 +78,174 @@ class CssWriter {
 	}
 }
 
-module.exports = function svelte(options = {}) {
+/** @returns {import('rollup').Plugin} */
+module.exports = function (options = {}) {
+	const extensions = options.extensions || ['.svelte'];
 	const filter = createFilter(options.include, options.exclude);
 
-	const extensions = options.extensions || ['.html', '.svelte'];
-
-	const fixed_options = {};
-
-	Object.keys(options).forEach(key => {
-		// add all options except include, exclude, extensions, and shared
-		if (pluginOptions[key]) return;
-		fixed_options[key] = options[key];
-	});
-
-	if (major_version >= 3) {
-		fixed_options.format = 'esm';
-		fixed_options.sveltePath = options.sveltePath || 'svelte';
-	} else {
-		fixed_options.format = 'es';
-		fixed_options.shared = require.resolve(options.shared || 'svelte/shared.js');
-	}
+	/** @type {import('svelte/types/compiler/interfaces').ModuleFormat} */
```

Comment:
seems like we should just make this a `.ts` file at some point :smile:

Reply:
Haha, I limited myself to 2 typedefs, just so that Intellisense would help out // stop complaining